remotes/docker/authorizer.go

```diff
@@ -109,15 +121,27 @@ func (a *dockerAuthorizer) Authorize(ctx context.Context, req *http.Request) err
 		return nil
 	}
 
-	auth, err := ah.authorize(ctx)
+	auth, refreshToken, err := ah.authorize(ctx)
 	if err != nil {
 		return err
 	}
 
 	req.Header.Set("Authorization", auth)
+	a.mu.Lock()
+	a.fetchedRefreshToken = refreshToken
+	a.mu.Unlock()
 	return nil
 }
 
+// RefreshToken returns a refresh token.
+// Needs to be called after Authorize().
+func (a *dockerAuthorizer) RefreshToken() string {
+	a.mu.Lock()
+	defer a.mu.Unlock()
```

Comment:
Maybe change this mutex to sync.RWMutex and use a.mu.RLock() here?

Reply:
fixed